documentation/modules/exploit/multi/http/apache_couchdb_erlang_rce.md

```diff
@@ -0,0 +1,260 @@
+## Vulnerable Application
+### Description
+This module exploits CVE-2022-24706, an unauthenticated RCE vulnerability in Apache CouchDB in versions 3.2.1 and below.
+
+Apache CouchDB is written in Erlang and so it has built-in support for distributed computing (clustering). The
+cluster nodes communicate using the Erlang/OTP Distribution Protocol, which provides for the possibility of executing
+OS command requests as the user running the software.
+
+In order to connect and run OS commands, one needs to know the secret phrase or in Erlang terms the "cookie". The CouchDB
+installer in versions 3.2.1 and below, by default, sets the cookie to "monster".
+
+### Setup
+#### Ubuntu 20.04
+Create a CouchDB user:
+```
+sudo adduser --system \
+             --home /opt/couchdb \
+             --no-create-home \
+             --shell /bin/bash \
+             --group --gecos \
+             "CouchDB Administrator" couchdb
+sudo passwd couchdb
+```
+
+Install dependencies:
+```
+sudo apt-get --no-install-recommends -y install \
+    build-essential pkg-config erlang erlang-reltool \
+    libicu-dev libmozjs-68-dev python3
+sudo apt-get -y install pip sphinx-doc sphinx-common
+sudo pip install --upgrade sphinx_rtd_theme nose requests hypothesis
+```
+
+Download vulnerable version of CouchDB:
+```
+wget https://downloads.apache.org/couchdb/source/3.2.1/apache-couchdb-3.2.1.tar.gz
+gunzip apache-couchdb-3.2.1.tar.gz
+tar -xvf apache-couchdb-3.2.1.tar
+```
+
+Copy the built couchdb release to the new user's home directory:
+`cp -R /path/to/couchdb/rel/couchdb /opt/couchdb`
+
+Change the ownership & permission of the CouchDB directories by running:
+```
+chown -R couchdb:couchdb /opt/couchdb
+find /opt/couchdb -type d -exec chmod 0770 {} \;
+```
+
+Update the permissions for the ini files:
+`chmod 0644 /opt/couchdb/etc/*`
+
+Change the bind address of [chttpd] and [httpd] in `</path/to/couchdb/>etc/default.ini` from 127.0.0.1 to 0.0.0.0 or
+or whatever you prefer
+```
+[httpd]
+port = 5986
+bind_address = 0.0.0.0
+[chttpd]
+; These settings affect the main, clustered port (5984 by default).
+port = 5984
+bind_address = 0.0.0.0
+```
+
+You can start the CouchDB server by running:
+`sudo -i -u couchdb couchdb/bin/couchdb`
+
+Navigate to `http://<IP-ADDRESS>:5984/_utils/` to verify the install
+
+#### Windows
+
+Download the following installer:
+`https://archive.apache.org/dist/couchdb/binary/win/2.3.1/apache-couchdb-2.3.1.msi`
+
+Click the installer to launch, accept the licensing agreement. CouchDB should now be installed.
+
+Change the bind address of [chttpd] and [httpd] in `</path/to/couchdb/>etc/default.ini` from 127.0.0.1 to 0.0.0.0 or
+or whatever you prefer
+```
+[httpd]
+port = 5986
+bind_address = 0.0.0.0
+[chttpd]
+; These settings affect the main, clustered port (5984 by default).
+port = 5984
+bind_address = 0.0.0.0
+```
+Navigate to `http://<IP-ADDRESS>:5984/_utils/` to verify the install
+
```

Comment:
Before this is done you need to restart the Apache CouchDB service so the new changes take effect